pulser-core/pulser/sequence/sequence.py

```diff
@@ -423,63 +420,63 @@ def switch_device(
-                        np.all(ch_sample_phase == ch_sample_phase[0])
-                    )
-                # Phase_jump_time and cloc_period check
+                    phase_is_constant = bool(np.all(
+                        ch_sample_phase == ch_sample_phase[0]
+                    ))
+                # Phase_jump_time and clock_period check
                 phase_jump_time_check = (
                     o_d_ch_obj.phase_jump_time == n_d_ch_obj.phase_jump_time
                 )
                 clock_period_check = (
                     o_d_ch_obj.clock_period % n_d_ch_obj.clock_period == 0
                 )
-                if phase_is_constant:
-                    if clock_period_check:
-                        channel_match[o_d_ch_name] = n_d_ch_id
-                        channel_match["alert_phase_jump"] = (
-                            phase_jump_time_check is False
-                        )
-                        break
-                    else:
-                        channel_match[o_d_ch_name] = None
-                        channel_match["strict_error_message"] = (
-                            "No channel with phase_jump_time"
-                            + " & clock_period match."
-                        )
-                    continue
-                if phase_jump_time_check and clock_period_check:
+                if clock_period_check and (
+                    phase_is_constant or phase_jump_time_check
+                ):
                     channel_match[o_d_ch_name] = n_d_ch_id
+                    alert_phase_jump = not phase_jump_time_check
                     break
                 else:
                     channel_match[o_d_ch_name] = None
-                    channel_match["strict_error_message"] = (
+                    strict_error_message = (
                         "No channel with phase_jump_time"
                         + " & clock_period match."
                     )
 
         if None in channel_match.values():
-            if channel_match["strict_error_message"] != "":
-                raise ValueError(channel_match["strict_error_message"])
+            if strict_error_message != "":
+                raise ValueError(strict_error_message)
             else:
-                raise TypeError(channel_match["ch_type_er_mess"])
+                raise TypeError(ch_type_er_mess)
         # Initialize the new sequence
         new_seq = Sequence(self.register, new_device)
 
         for call in self._calls[1:]:
             if not (call.name == "declare_channel"):
                 getattr(new_seq, call.name)(*call.args, **call.kwargs)
                 continue
-            sw_channel_args = list(call.args)
             # Switch the old id with the correct id
-            sw_channel_args[1] = channel_match[sw_channel_args[0]]
-            if strict and channel_match["alert_phase_jump"]:
+            sw_channel_args = list(call.args)
+            sw_channel_kw_args = call.kwargs
```

Comment:
You shouldn't modify this `dict` in place, since it will affect the original. 
```suggestion
            sw_channel_kw_args = call.kwargs.copy()
```